docs/BEMAN_STANDARD.md

```diff
@@ -140,6 +140,22 @@ contain a one- or two-paragraph summary describing the library's purpose.
 **[CMAKE.PROJECT_NAME]** RECOMMENDATION: The CMake project name should be
 identical to the beman library name.
 
+**[CMAKE.PASSIVE_PROJECTS]** REQUIREMENT: CMake projects must not adjust
+user-specified compilation flags.
+
+User-provided compilation flags, whether specified via presets, command-line
+options, or toolchains, must not be modified by CMake projects. Therefore, CMake
+projects may not set variables that impact compilation flags such as
+`CMAKE_CXX_FLAGS` and `CMAKE_CXX_STANDARD`, as this would override user settings.
+
+For common compiler/flag combinations, it is recommended to provide CMake presets
+as a convenient alternative for users.
+
+If specific compiler flags are essential for project functionality (e.g., C++
+standard features), use utilities like `check_cxx_source_compiles` to detect
```

Comment:
Is there another way that does not conflict with the rules introduced here that can uniformlly test C++ standard support?

It make sense a library will work with assumptions of "I can take advantage of a few language feature optionally, but I have a minimum targeted C++ version".

It will be error prone to test every individual language feature using `check_cxx_source_compiles` and might make #58 impossibly tedious to implement.

Reply:
It might make sense to create a requirement for order of trust for `check_cxx_source_compiles`, e.g. feature macro > testing if class exist > testing behavior of code.

For example, trivial union is already the accepted behavior for some compilers (even under constexpr) before the adoption of paper. If a library author checks for trivial union by behavior instead of feature flag, this maybe undefined behavior inside of the compiler instead of an actual paper conforming implementation, which should not be relied on and may be inconsistent.

e.g. quoting the paper:
> The problem is that the above does not work (although there is [implementation divergence](https://godbolt.org/z/a3318n63v) - MSVC and EDG accept it and GCC did accept it even up to 13.2, but GCC trunk and Clang reject).

With this order of trust in-place, we would be able to ask implementation author to move to feature test macro as soon as its stabilized. 

Or we can make it a recommendation/ requirement to disable non standardized feature by default and switch to `check_cxx_source_compiles` once the feature is standardized with stable compiler support. However I don't think this would be a good approach.

Reply:
> It will be error prone to test every individual language feature using check_cxx_source_compiles for version testing

One can use the `__cplusplus` macro instead of feature test macros if so desired from within `check_cxx_source_compiles`.

Reply:
Oh cool I didn't know this is standard!